app/forms/idv/api_image_upload_form.rb

```diff
@@ -439,42 +459,47 @@ def store_failed_images(client_response, doc_pii_response)
         return {
           front: [],
           back: [],
+          selfie: [],
         }
       end
       # doc auth failed due to non network error or doc_pii is not valid
       if client_response && !client_response.success? && !client_response.network_error?
         errors_hash = client_response.errors&.to_h || {}
-        ## assume both sides' error presents or both sides' error missing
-        failed_front_fingerprint = extra_attributes[:front_image_fingerprint]
-        failed_back_fingerprint = extra_attributes[:back_image_fingerprint]
-        ## not both sides' error present nor both sides' error missing
-        ## equivalent to: only one side error presents
-        only_one_side_error = errors_hash[:front]&.present? ^ errors_hash[:back]&.present?
-        if only_one_side_error
-          ## find which side is missing
-          failed_front_fingerprint = nil unless errors_hash[:front]&.present?
-          failed_back_fingerprint = nil unless errors_hash[:back]&.present?
+        failed_front_fingerprint = nil
+        failed_back_fingerprint = nil
+        if errors_hash[:front] || errors_hash[:back]
+          if errors_hash[:front]
+            failed_front_fingerprint = extra_attributes[:front_image_fingerprint]
+          end
+          if errors_hash[:back]
+            failed_back_fingerprint = extra_attributes[:back_image_fingerprint]
+          end
+        elsif !client_response.doc_auth_success?
+          failed_front_fingerprint = extra_attributes[:front_image_fingerprint]
+          failed_back_fingerprint = extra_attributes[:back_image_fingerprint]
         end
-        document_capture_session.
-          store_failed_auth_data(
-            front_image_fingerprint: failed_front_fingerprint,
-            back_image_fingerprint: failed_back_fingerprint,
-            doc_auth_success: client_response.doc_auth_success?,
-            selfie_status: selfie_status_from_response(client_response),
-          )
+        document_capture_session.store_failed_auth_data(
+          front_image_fingerprint: failed_front_fingerprint,
+          back_image_fingerprint: failed_back_fingerprint,
+          selfie_image_fingerprint: extra_attributes[:selfie_image_fingerprint],
```

Comment:
It seems odd to me that we have `failed_front_fingerprint` and `failed_back_fingerprint` then `extra_attributes[:selfie_image_fingerprint]`. I'm not saying that using `extra_attributes` is wrong, more that I'd expect we're doing the same things with all three images and I'm not sure why the selfie is different?

Reply:
hey @charleyf 👋🏿  - the doc auth result errors exist across 2 granular parts (front and back image) and the front and failed_(front|back)_fingerprints are defined conditionally based on doc auth response errors that may exist for the front or back images.  here we are  are always passing the fingerprint to the the document capture session

Reply:
Ah. The reason I was asking is that it felt weird to me that we needed semi-complex conditional logic for the `front` and `back` and not the `selfie`.

The more I think about this, the more I'm pretty sure that the conditional logic here needs to change to reflect how we want to handle `selfie_fingerprint` errors.

My reasoning:
- `if client_response && !client_response.success? && !client_response.network_error?` is checking "Did the client response fail in a way where we should store the image fingerprints as 'failed' ?"
  - This means that when selfie is enabled, we'll always store the selfie fingerprint, regardless if it was the selfie that failed. This is where I'd expect a check on the `:selfie_status === :Fail`
- `elsif doc_pii_response && !doc_pii_response.success?`
   - This says "If the doc_pii_response fails"
   - `selfie_image_fingerprint: extra_attributes[:selfie_image_fingerprint],` then mark the selfie as failed.
   - I don't think that in this case we do actually want to fail the selfie? The PII is not affected by the selfie?

Reply:
we are always passing the selfie fingerprint but [conditionally store the selfie_fingerprint](https://github.com/18F/identity-idp/blob/19e47acc6f22559d296b0e2cc1243e26506600f9/app/models/document_capture_session.rb#L43)

Reply:
Interesting, I missed that. Does it make sense to match `failed_front_fingerprint` and `back_image_fingerprint` and do something like this in `store_failed_images`?

```ruby
      failed_selfie_fingerprint = nil
      if client_response && client_response.selfie_status == :Fail
          failed_selfie_fingerprint = extra_attributes[:selfie_image_fingerprint]
      end
```

Reply:
yes, i did think about that 🤔 

my reasoning for putting this condition in the DocumentCaptureSession was that if we call store_failed_auth_data in more places in the future it would require us to remember and duplicate this condition/code. however, in regards to the front and back fingerprint values, the business logic for storing front and back images is pretty specific to the errors here and also can happen if doc auth result fails while we have no errors returned from the vendor.

Reply:
That makes sense, I would still expect the conditionals that determine when we store fingerprints to be in one place: either `store_failed_images` or `DocumentCaptureSession`, but I may be misunderstanding a benefit here.

Either of these would make more sense to me in the session 🤷 
```ruby
session_result.add_failed_front_image!(front_image_fingerprint)  if front_image_fingerprint
session_result.add_failed_back_image!(back_image_fingerprint)  if back_image_fingerprint
session_result.add_failed_selfie_image!(selfie_image_fingerprint) if selfie_image_fingerprint
```

```ruby
session_result.add_failed_front_image!(front_image_fingerprint)
session_result.add_failed_back_image!(back_image_fingerprint)
session_result.add_failed_selfie_image!(selfie_image_fingerprint)
```